src/main/java/org/cactoos/io/TailInput.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.cactoos.Input;
+import org.cactoos.text.FormattedText;
+
+/**
+ * Input showing only last N bytes of the stream.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
+ * @version $Id$
+ * @since 0.30
+ */
+public final class TailInput implements Input {
+
+    /**
+     * Input to decorate.
+     */
+    private final Input input;
+
+    /**
+     * Number of last bytes to show from the input.
+     */
+    private final int count;
+
+    /**
+     * Maximum number of bytes to read at once.
+     */
+    private final int max;
```

Comment:
@krzyk why bother parametrizing this?

Reply:
@llorllale All other cases where we read to a buffer are parameterized. And it is useful for testing, you don't need to share (either publicly or in package) a constant, instead you provide the data right in the constructor - better encapsulation.